src/lizaalert/courses/models.py

```diff
@@ -418,6 +418,54 @@ class Meta:
         ordering = ("knowledge",)
 
 
+class Cohort(TimeStampedModel):
+    """
+    Модель для представления группы курса.
+
+    Поля модели:
+    - course: ForeignKey к модели Course
+    - cohort_number: уникальный номер группы в рамках курса
+    - start_date: дата начала обучения
+    - end_date: дата окончания обучения
+    - students_count: количество студентов в группе
+    - teacher: имя преподавателя группы
+    """
+
+    course = models.ForeignKey(Course, on_delete=models.PROTECT, related_name="cohorts", verbose_name="Курс")
+    cohort_number = models.PositiveIntegerField(
+        verbose_name="Номер группы", help_text="Данное поле будет рассчитано автоматически при сохранении."
+    )
+    start_date = models.DateField(verbose_name="Дата начала", null=True, blank=True)
+    end_date = models.DateField(verbose_name="Дата окончания", null=True, blank=True)
+    students_count = models.PositiveIntegerField(verbose_name="Количество студентов", null=True, blank=True)
+    teacher = models.ForeignKey(User, on_delete=models.PROTECT, verbose_name="Преподаватель")
+
+    class Meta:
+
+        constraints = [
+            models.UniqueConstraint(
+                fields=["course", "cohort_number"],
+                name="unique_course_cohort_number",
+            )
+        ]
+        verbose_name = "Группа курса"
+        verbose_name_plural = "Группы курса"
+
+    def save(self, *args, **kwargs):
+        if not self.pk:
+            max_cohort_number = Cohort.objects.filter(course=self.course).aggregate(Max("cohort_number"))[
+                "cohort_number__max"
+            ]
+
+            self.cohort_number = max_cohort_number + 1 if max_cohort_number is not None else 1
+
+        with transaction.atomic():
+            super().save(*args, **kwargs)
+
+    def __str__(self):
+        return f"{self.course.title} - Группа {self.cohort_number}"
```

Comment:
```suggestion
        return f"{self.course_id} - Группа {self.cohort_number}"
```

не нужно делать циклических запросов